### PR TITLE
Add limit/offset to .all() queries

### PR DESCRIPTION
**Pull request by mflaxman**

### CODE REVIEW

File: subconscious/model.py [hunk 1]
```diff
@@ -265,8 +265,19 @@ def get_index_key(cls, column_name):
         return all_keys
 
     @classmethod
-    async def all(cls, db, order_by=None):
-        for redis_key in await cls._get_ids_for_all(db, order_by=order_by):
+    async def all(cls, db, order_by=None, limit=None, offset=None):
+        ids_to_iterate = await cls._get_ids_for_all(db, order_by=order_by)
+        if order_by:
```
Comment (gipsy86147) [inline subconscious/model.py]: Do we really want to force the user to use 'orderby' in order to use the limit/offset?
What if the user call it without 'orderby'  ? User will get unexpected number of records back as we are disregarding 'limit' and 'offset' in this scenario.

Comment (mflaxman) [reply subconscious/model.py]: My thinking is what does limit/offset mean without order by? Perhaps you could do limit without order by (effectively random order), but an offset without an order just makes no sense.

How about we keep it a requirement for now but we can always relax it in the future?

Comment (gipsy86147) [reply subconscious/model.py]: Then make it fail if called with no 'orderby'

Comment (mflaxman) [reply subconscious/model.py]: Ya, that's yucky too. I just pushed a PR to allow it all since it's technically not wrong (just strange).

**Pull request merged**